### ISSUE TITLE
Failed to patch the runtime: Permission denied (os error 13)

### ISSUE THREAD

**Issue by AlexDumanskiy** [filips123/PWAsForFirefox#309]
I realize that the problem is probably on my side, but I have no idea which exact permissions are missing.
I would suggest improving the error message or log to include details, so the user will be able to fix this.

Gentoo Linux, native part installed from sources, tag v2.4.1
Followed all instruction (I think), plugin finished installation, downloaded runtime, no additional options selected.

Directory ~/.local/share/firefoxpwa/runtime/ has runtime files, they are owned by my user and are definitely writable.
Log file has this:
```
22:44:19 [ERROR] Failed to process request

Caused by:
    0: Failed to patch the runtime
    1: Permission denied (os error 13)

```
JS console does not show anything of interest.

**Comment by AlexDumanskiy** [comment]
Double-checked that the runtime is writable by the user:

```
find ~/.local/share/firefoxpwa -exec touch '{}' \; ; echo $?
0

```

**Comment by camillemndn** [comment]
Hi, I had the same error and for me it had to do with two files that are rewritten at each launch of firefoxpwa:
```chmod u+w $HOME/.local/share/firefoxpwa/runtime/_autoconfig.cfg```
and
```chmod u+w $HOME/.local/share/firefoxpwa/runtime/defaults/pref/autoconfig.js ```.
But it would be better to understand why firefoxpwa wants to rewrite these files.

**Comment by AlexDumanskiy** [comment]
@camillemndn FYI your patch from here: https://github.com/filips123/PWAsForFirefox/pull/311 did not fix the issue on my system.

I'll try using auditd to figure where the addon tries to gain access to later.

**Comment by AlexDumanskiy** [comment]
@camillemndn thanks for you help!

When I do `firefoxpwa runtime install` it seems okay:
```
10:00:46 [0] ~ >firefoxpwa runtime install                                   
09:01:32 [WARN] This will download the unmodified Mozilla Firefox and locally modify it
09:01:32 [WARN] Firefox is licensed under the Mozilla Public License 2.0
09:01:32 [WARN] Firefox is a trademark of the Mozilla Foundation in the U.S. and other countries
09:01:32 [WARN] This project is not affiliated with the Mozilla Foundation in any way
09:01:32 [WARN] By using this project you also agree to the Firefox Privacy Notice: https://www.mozilla.org/privacy/firefox/
09:01:32 [WARN] Check the Firefox website for more details: https://www.mozilla.org/firefox/
09:01:32 [INFO] Downloading the runtime archive
09:01:37 [INFO] Extracting the runtime archive
09:01:43 [INFO] Copying the runtime
09:01:44 [INFO] Runtime installed!
10:01:44 [0] ~ >          
```

**Comment by camillemndn** [comment]
Ok, and do you have more verbose log if you do ```firefoxpwa site launch <SITEID>```?

**Comment by AlexDumanskiy** [comment]
I'm sorry, I just can't ignore that point - it has a space in the binary name? Wow. Never thought I'd see that.

**Comment by camillemndn** [comment]
What do you mean? Isn't the binary named ```firefoxpwa``` haha?

**Comment by AlexDumanskiy** [comment]
Logs didn't get more verbose:

```
10:05:09 [0] ~ >firefoxpwa site launch 01GTWGQQVBTS5K29R5V4M08WWY                                      
09:05:20 [INFO] Patching the runtime
09:05:20 [ERROR] Failed to patch the runtime

Caused by:
    Permission denied (os error 13)
10:05:20 [0] ~ >            
```

**Comment by AlexDumanskiy** [comment]
@camillemndn, 

> 

Sorry my bad, I misread it.

**Comment by camillemndn** [comment]
Can you ```ls -allR /usr/share/firefoxpwa```?

**Comment by camillemndn** [comment]
Ok, well there you have it! :D

**Comment by AlexDumanskiy** [comment]
Nice, thank you a lot!

**Comment by filips123** [comment]
This issue can be closed because the original problem has been solved and the other one will be soved with #311, right?

**Comment by camillemndn** [comment]
> This issue can be closed because the original problem has been solved and the other one will be soved with #311, right?

Yes, it was solved.
